File: src/main/java/com/project/budgetapp/models/Category.java
package com.project.budgetapp.models;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Entity(name="categories")
public class Category {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long category_id;
    private String name;
    private long account_id;


    public Category() {}

    public long getCategory_id() {
        return category_id;
    }

    public void setCategory_id(long category_id) {
        this.category_id = category_id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public long getAccount_id() {
        return account_id;
    }

    public void setAccount_id(long account_id) {
        this.account_id = account_id;
    }
}

File: Front-end/src/app/models/category.ts
export class Category {
    category_id: number;
    category_name: string;
    account_id: number;

    constructor(category_id: number,
        category_name: string,
        account_id: number) {
            this.account_id = account_id;
            this.category_id = category_id;
            this.category_name = category_name;
        }
}